x-pack/plugins/security/server/authentication/authentication_result.ts

```diff
@@ -62,14 +63,22 @@ export class AuthenticationResult {
   /**
    * Produces `AuthenticationResult` for the case when authentication succeeds.
    * @param user User information retrieved as a result of successful authentication attempt.
+   * @param [authHeaders] Optional dictionary of the HTTP headers with authentication information.
    * @param [state] Optional state to be stored and reused for the next request.
    */
-  public static succeeded(user: AuthenticatedUser, state?: unknown) {
+  public static succeeded(
```

Comment:
Actually, that's how I [did it in the beginning](https://github.com/elastic/kibana/pull/39446/commits/3cff787f7169575058c7649c6700ddbee0a1fbbe#diff-7383cd57386800ed57a48d81986248f8R69), but eventually abandoned this idea as it became hard to read and interpret, e.g.:
```
return AuthenticationResult.succeeded(user, { authorization }, { authorization });
```
In the snippet above we pass both `authHeaders` and `state` and it's hard to tell what we're doing here exactly. Also TS won't complain if we decide to switch the order we pass `state` and `authHeaders` in since types of the args are allowing that (and I managed to spend some _unreasonable_ amount of time looking at the code and trying to understand what was wrong when I did that :see_no_evil: ).

So I'm leaning towards keeping it as is, but would like to hear what you think?

Reply:
That makes sense... let's leave it how it is then.